metrics.go

```diff
@@ -103,45 +109,57 @@ func (c *aggregatingCollector) Describe(descs chan<- *prometheus.Desc) {
 	}
 }
 
-func (c *aggregatingCollector) Collect(metrics chan<- prometheus.Metric) {
-	now := time.Now()
-	c.mutex.Lock()
-	var segsSent, segsRcvd uint64
-	var bytesSent, bytesRcvd uint64
-	for _, conn := range c.conns {
-		info, err := conn.getTCPInfo()
-		if err != nil {
-			if strings.Contains(err.Error(), "use of closed network connection") {
-				c.closedConn(conn)
+func (c *aggregatingCollector) cron() {
+	ticker := time.NewTicker(collectFrequency)
+	defer ticker.Stop()
+
+	for now := range ticker.C {
+		c.mutex.Lock()
+		c.segsSent = 0
+		c.segsRcvd = 0
+		c.bytesSent = 0
+		c.bytesRcvd = 0
+		for _, conn := range c.conns {
+			info, err := conn.getTCPInfo()
+			if err != nil {
+				if strings.Contains(err.Error(), "use of closed network connection") {
+					c.closedConn(conn)
+					continue
+				}
+				log.Errorf("Failed to get TCP info: %s", err)
 				continue
 			}
```

Comment:
Is it a problem that if we `conn.getTCPInfo()` errors differently from expected then we'll never close the connection, right?

I've seen `2021-07-08T21:39:47.925-0400    ERROR   tcp-tpt go-tcp-transport@v0.2.2/metrics.go:118  Failed to get TCP info: raw-control tcp 192.168.1.6:4001: getsockopt: not implemented` on Windows so I'm concerned this could cause us problems.

Reply:
I tried it in my Windows VM, and I'm getting the same error. Should we disable metrics collection on Windows?